NineChroniclesUtilBackend.Store/Scrapper/ArenaScrapper.cs

```diff
@@ -7,92 +7,36 @@
 
 namespace NineChroniclesUtilBackend.Store.Scrapper;
 
-public class ArenaScrapper
+public class ArenaScrapper(ILogger<ArenaScrapper> logger, IStateService service, EmptyChronicleClient client, MongoDbStore store)
 {
-    private readonly ILogger<ArenaScrapper> _logger;
-    public readonly ScrapperResult Result = new ScrapperResult();
+    private readonly ILogger<ArenaScrapper> _logger = logger;
 
-    private StateGetter _stateGetter;
-    private EmptyChronicleClient _client;
-    public event EventHandler<ArenaDataCollectedEventArgs> OnDataCollected;
-    
-    public ArenaScrapper(ILogger<ArenaScrapper> logger, IStateService service, EmptyChronicleClient client)
-    {
-        _stateGetter = new StateGetter(service);
-        _logger = logger;
-        _client = client;
-    }
-
-    protected virtual void RaiseDataCollected(ArenaData arenaData, AvatarData avatarData)
-    {
-        OnDataCollected?.Invoke(this, new ArenaDataCollectedEventArgs(arenaData, avatarData));
-    }
+    private readonly IStateService _stateService = service;
+    private readonly EmptyChronicleClient _client = client;
+    private readonly MongoDbStore _store = store;
```

Comment:
Maybe it's just that I'm not familiar with the C# event function, but I thought it increased complexity. In the existing code, MongoDBStore and ArenaScrapper were connected by injecting a handler in the Worker through the event function. It is put in bulk, but since MongoDBStore does not have the same context as ArenaScrapper, it was written to randomly execute the FlushAsync method to put the remaining values into the DB. This was a bad pattern for the polling logic we would write later. When polling, values must be updated in block units and immediately reflected in the database, but this was no different from providing the Bulk method.

I sympathize with the case of replacing the database separately. I think this can be isolated with the repository interface.